client/my-sites/domains/domain-management/list/all-domains.jsx

```diff
@@ -409,11 +409,14 @@ class AllDomains extends Component {
 			} );
 		}
 
+		const queryParams = new URLSearchParams( window.location.search );
+
 		return (
 			<>
 				<div className="all-domains__filter">{ this.renderDomainTableFilterButton() }</div>
 				<DomainsTable
 					currentRoute={ currentRoute }
+					defaultSortKey={ queryParams.get( 'sortKey' ) }
```

Comment:
I don't think the `all-domains` component should be aware of the sorting. I'd probably read the query param inside the table and remove the prop.